autogen/agentchat/contrib/retriever/base.py

```diff
@@ -0,0 +1,85 @@
+from abc import ABC, abstractmethod
+from typing import List, Union, Callable, Any
+
+
+class Retriever(ABC):
+    def __init__(
+        self,
+        path="./db",
+        name="vectorstore",
+        embedding_model_name="all-MiniLM-L6-v2",
+        embedding_function=None,
+        max_tokens: int = 4000,
+        chunk_mode: str = "multi_lines",
+        must_break_at_empty_line: bool = True,
+        custom_text_split_function: Callable = None,
+        client=None,
+        # TODO: add support for custom text types and recurisive
+        custom_text_types: str = None,
+        recursive: bool = True,
+    ):
+        """
+        Args:
+            path: path to the folder where the database is stored
+            name: name of the database
+            embedding_model_name: name of the embedding model to use
+            embedding_function: function to use to embed the text
+            max_tokens: maximum number of tokens to embed
+            chunk_mode: mode to chunk the text. Can be "multi_lines" or "single_line"
+            must_break_at_empty_line: chunk will only break at empty line if True. Default is True.
+                    If chunk_mode is "one_line", this parameter will be ignored.
+            custom_text_split_function: custom function to split the text into chunks
```

Comment:
Not all parameters explained in the docstring

Reply:
I don't see the docstrings for "custom_text_types" and "recursive"

Reply:
I have added them to base class member functions. But they are not being used by the DBs yet. It was yet another change in main which I didn't know where to insert

Reply:
Added them in docstring and also in the instance variable list. They are accessible to the DBs but are not being used currently-- These 2 were again changes from upstream that I didn't have context about so just added them as TODOs earlier